CMakeLists.txt

```diff
@@ -165,56 +176,107 @@ if (CMAKE_BUILD_TYPE STREQUAL "Release")
+    "/wd4551" # Suppress warnings about function call missing argument list
+    "/w"      # disable other warnings
+    "/FS"
+    "/GS"       # Enable buffer security checks
+    "/guard:cf" # Enable control flow guard (security feature)
+    "/EHsc"
+  )
+  add_definitions(-DMS_WINDOWS)  # Globally define MS_WINDOWS
+  else()
   add_compile_options(
-    "-g"
-    "-O1"
-    "-fno-omit-frame-pointer"
+    "-Wno-c99-designator"         # Check if still necessary?
+    "-Wno-return-type-c-linkage"  # Check if still necessary?
+    "-Wno-macro-redefined"        # Check if still necessary?
+    "-Wno-implicit-const-int-float-conversion" # Necessary for DataSketches
+    "-fwrapv"                     # Conda used to force-add. Remove in follow up PR
+    "-fstack-protector-strong"    # Conda used to force-add. Remove in follow up PR
+    "-D_FORTIFY_SOURCE=2"         # Conda used to force-add. Remove in follow up PR
   )
 endif()
 
+# Debug Compiler Args
+if (CMAKE_BUILD_TYPE STREQUAL "Debug" OR CMAKE_BUILD_TYPE STREQUAL "DebugSanitize")
+  if (NOT WIN32)
+    add_compile_options(
+      "-g"
+      "-O1"
+      "-fno-omit-frame-pointer"
+    )
+  else()
+    add_compile_options(
+      "/Zi"  # Generate debug information
+    )
+    add_link_options("/DEBUG") # Generate debug symbols
+  endif()
+endif()
+
 # Debug with Sanitizer Flags
+# TODO [BSE-4552]: Add DebugSanitize as a build type for Windows builds.
 if (CMAKE_BUILD_TYPE STREQUAL "DebugSanitize")
-  add_compile_options(
-    "-fsanitize=address"
-    "-fsanitize=undefined"
-  )
-  add_link_options(
-    "-fsanitize=address"
-    "-fsanitize=undefined"
-  )
+  if (NOT WIN32)
+    add_compile_options(
+      "-fsanitize=address"
+      "-fsanitize=undefined"
+    )
+    add_link_options(
+      "-fsanitize=address"
+      "-fsanitize=undefined"
+    )
+  else()
+    # Note for DebugSanitize builds we have to compile with /bigobj to allow larger object files
+    # https://learn.microsoft.com/en-us/cpp/build/reference/bigobj-increase-number-of-sections-in-dot-obj-file?view=msvc-170
+    add_compile_options(
+      "/bigobj"
+      "/fsanitize=address"
+    )
+    # These definitions are necessary for now because of conflicts between our extensions and other
+    # build dependencies e.g. fmt. In the future we can look into enabling them:
+    # https://learn.microsoft.com/en-us/cpp/sanitizers/error-container-overflow?view=msvc-170
+    # TODO [BSE-4552] Remove once we have fully working DebugSanitize config
+    add_definitions(
+      -D_DISABLE_STRING_ANNOTATION
+      -D_DISABLE_VECTOR_ANNOTATION
+    )
+  endif()
 endif()
 
 # Release Compiler Args
 if (CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
-  add_compile_options(
-    "-O3"
-    "-ftree-vectorize"
-    # Encourages LTO & increase obfuscations
-    $<$<COMPILE_LANGUAGE:CXX>:-fvisibility=hidden>
-  )
+  if (NOT WIN32)
+    add_compile_options(
+      "-O3"
```

Comment:
Couldn you add a todo to this, cause we want to make sure we are compiling with the highest optimizations on releases.